tests/testautorelay.nim

```diff
@@ -77,13 +77,13 @@ suite "Autorelay":
     await switchClient.connect(switchRelay.peerInfo.peerId, switchRelay.peerInfo.addrs)
     await fut.wait(1.seconds)
     let addresses = autorelay.getAddresses()
+
     check:
-      addresses == @[buildRelayMA(switchRelay, switchClient)]
-      addresses.len() == 1
-      addresses[0] in switchClient.peerInfo.addrs
-    await allFutures(switchClient.stop(), switchRelay.stop())
+      addresses == buildRelayMA(switchRelay, switchClient)
+    for address in addresses:
+      check address in switchClient.peerInfo.addrs
 
-    check addresses != switchClient.peerInfo.addrs
+    await allFutures(switchClient.stop(), switchRelay.stop())
 
   asyncTest "Three relays connections":
```

Comment:
I'll answer every question about this test here.
So. This badly named "Three relays connections" tests multiple things. The two big things tested is:
- the maximum number of relay line 119: `let autorelay = AutoRelayService.new(2, relayClient, checkMA, newRng())` (here it's a limit of 2)
- what happens in case of a disconnection of a remote peer line 127 `await rel2.stop()`

What does this test do in details:
- It creates 4 switches. switchClient is the "autorelay switch", the other three (rel1, rel2, rel3) are standard relay server switches.
- State == 0: checkMA is called for the first time because switchClient connects to rel1
- State == 1: switchClient connects to rel2 (line 124). You can notice that switchClient connects to rel3 aswell but it doesn't do anything du to the maximum number of relay set to 2.
- State == 2: rel2 disconnect, we return to the "first state" with only rel1 connected to switchClient
- State == 3: now that rel2 is disconnected, there's room for another relay, and rel3 takes this place.

Reply:
Thanks a lot.